GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m

```diff
@@ -1002,6 +1064,89 @@ + (void)proxyAppDelegate:(id<GULApplicationDelegate>)appDelegate {
   }
 }
 
+#if UISCENE_SUPPORTED
++ (void)proxySceneDelegate:(UIScene *)scene {
+  Class realClass = [scene.delegate class];
+
+  // Skip proxying if the class has a prefix of kGULAppDelegatePrefix, which means it has been
+  // proxied before.
+  if ([NSStringFromClass(realClass) hasPrefix:kGULAppDelegatePrefix]) {
+    return;
+  }
+
+  NSString *classNameWithPrefix =
+      [kGULAppDelegatePrefix stringByAppendingString:NSStringFromClass(realClass)];
+  NSString *newClassName =
+      [NSString stringWithFormat:@"%@-%@", classNameWithPrefix, [NSUUID UUID].UUIDString];
+
+  if (NSClassFromString(newClassName)) {
+    GULLogError(kGULLoggerSwizzler, NO,
+                [NSString stringWithFormat:@"I-SWZ%06ld",
+                                           (long)kGULSwizzlerMessageCodeAppDelegateSwizzling005],
+                @"Cannot create a proxy for App Delegate. Subclass already exists. Original Class: "
+                @"%@, subclass: %@",
+                NSStringFromClass(realClass), newClassName);
+    return;
+  }
+
+  // Register the new class as subclass of the real one. Do not allocate more than the real class
+  // size.
+  Class sceneDelegateSubClass = objc_allocateClassPair(realClass, newClassName.UTF8String, 0);
+  if (sceneDelegateSubClass == Nil) {
+    GULLogError(kGULLoggerSwizzler, NO,
+                [NSString stringWithFormat:@"I-SWZ%06ld",
+                                           (long)kGULSwizzlerMessageCodeAppDelegateSwizzling006],
+                @"Cannot create a proxy for App Delegate. Subclass already exists. Original Class: "
+                @"%@, subclass: Nil",
+                NSStringFromClass(realClass));
+    return;
+  }
+
+  NSMutableDictionary<NSString *, NSValue *> *realImplementationsBySelector =
+      [[NSMutableDictionary alloc] init];
+
+  // For scene:openURLContexts:
+  SEL openURLContextsSEL = @selector(scene:openURLContexts:);
+  [self proxyDestinationSelector:openURLContextsSEL
+      implementationsFromSourceSelector:openURLContextsSEL
+                              fromClass:[GULAppDelegateSwizzler class]
+                                toClass:sceneDelegateSubClass
+                              realClass:realClass
+       storeDestinationImplementationTo:realImplementationsBySelector];
+
+  // Store original implementations to a fake property of the original delegate.
+  objc_setAssociatedObject(scene.delegate, &kGULRealIMPBySelectorKey,
```

Comment:
nit: It think readability of the class may be slightly improved if we create a "set" counterpart for `originalImplementationForSelector: object:` method and use it everywhere instead of referring to `kGULRealIMPBySelectorKey` directly. It may be a subject of a followup PR.

Reply:
I feel like using `kGULRealIMPBySelectorKey` may remind people that it's a runtime associated object and people will be more careful. While you suggestion totally makes sense. We can discuss more in a followup pr.